README.md

```diff
@@ -46,9 +46,16 @@ Just use your package managers as you usually would. Run `yarn install` in Yarn
 
 ## Known Good Releases
 
-When running Yarn or pnpm within projects that don't list a supported package manager, Corepack will default to a set of Known Good Releases. In a way, you can compare this to Node.js, where each version ships with a specific version of npm.
+When running Corepack within projects that don't list a supported package
+manager, it will default to a set of Known Good Releases. In a way, you can
+compare this to Node.js, where each version ships with a specific version of npm.
 
-The Known Good Releases can be updated system-wide using the `--activate` flag from the `corepack prepare` and `corepack hydrate` commands.
+If there is no Known Good Release for the requested package manager, Corepack
```

Comment:
In what case would there be no Known Good Release?

Reply:
Upon the first time Corepack is run with a given package manager. The idea behind this PR would be not to rely on what version is set on `config.json` but instead get the most up-to-date version from npm registry.

Reply:
How often does it check the registry? Does it check every time you invoke yarn/pnpm/npm?

Reply:
No, only the first time, at which point it saves it as Known Good Release. Users can then update (or downgrade) that version by using `corepack prepare` or `corepack hydrate` (e.g. `corepack prepare pnpm@latest --activate` to get the latest pnpm by default). We could make an automated process for the Known Good Version to expire after some time, but that's not part of this initial implementation.

Reply:
Thanks, I understand now 👍

I think expiring would could be confusing. I also think this any auto update could be confusing because it no longer ties the package manager version to the corepack version (or node version for that matter). 

I think it’s best to let the user decide when to upgrade and perhaps provide better docs how to upgrade (this isn’t super clear today). Without corepack, this is handled with a notification that says something like npm i -g npm@latest. 

Has any of this behavior been reviewed by the npm team? I think they had some ideas about Know Good Release.

Reply:
> I also think this any auto update could be confusing because it no longer ties the package manager version to the corepack version (or node version for that matter).

Well that's the reason Corepack exists, because the Node.js TSC doesn't want to have package managers tied in with a specific Node.js version – specifically, several Node.js TSC members have expressed that the reason they are suportive on Corepack is to get rid of having to do security releases every time a vulnerability is reported in npm rather than `node` itself.
So the auto-lookup during the first use _has to be_ the default behavior.


> I think it’s best to let the user decide when to upgrade and perhaps provide better docs how to upgrade (this isn’t super clear today). Without corepack, this is handled with a notification that says something like npm i -g npm@latest.

Yes that makes sense, letting the end-user in control is very much the goal here. So instead of `npm i -g npm@latest`, users would type `corepack prepare npm@latest --activate`, which is arguably quite similar. Improving docs is of course always a goal.

> Has any of this behavior been reviewed by the npm team? I think they had some ideas about Know Good Release.

The npm team hasn't shown much interest in the development of Corepack, so I don't think they have.